spec/p2p/v0.34/switch.md

```diff
@@ -0,0 +1,237 @@
+# Switch
+
+The switch is a core component of the p2p layer.
+It manages the procedures for [dialing peers](#dialing-peers) and
+[accepting](#accepting-peers) connections from peers, which are actually
+implemented by the [transport](./transport.md).
+It also manages the reactors, i.e., protocols implemented by the node that
+interact with its peers.
+Once a connection with a peer is established, the peer is [added](#add-peer) to
+the switch and all registered reactors.
+Reactors may also instruct the switch to [stop a peer](#stop-peer), namely
+disconnect from it.
+The switch, in this case, makes sure that the peer is removed from all
+registered reactors.
+
+## Dialing peers
+
+Dialing a peer is implemented by the `DialPeerWithAddress` method.
+
+This method is invoked by the [peer manager](./peer_manager.md#ensure-peers)
```

Comment:
I think these high level flows such as this one (`peer manager` (`DialPeer..`.) -> `Switch` -> `Transport`) would be very valuable to understanding the P2P layer. I wonder if the spec should not provide either a diagram, or simply list the essential 2-3 high level flows (such as the DialPeer above) that the P2P layer is responsible for.

Reply:
Producing something like that is one of the goals for Q4.